### PR TITLE
Reorganize classes for 2.0

### PR DESCRIPTION
**Pull request by jwage**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | #682

#### Summary

As we refactored, several new classes were created and put in the root namespace. Move the logical groupings in to folders.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,5 @@ phpunit.xml
 /phpcs.xml
 /.phpcs-cache
 /box.phar
+/.idea
```
Comment (morozov) [inline .gitignore]: IMO, this line belongs to a personal `.gitignore` file but not to the project, unless the project is tightly coupled to the IDE.

Comment (jwage) [reply .gitignore]: I didn't know you could do that. I will remove it and do that. Thanks!

Comment (jwage) [reply .gitignore]: Removed here https://github.com/doctrine/migrations/commit/d3d7a8ce0e5133c7c48b46cbaf9ee0ee48191492

Comment (Majkl578) [reply .gitignore]: > IMO, this line belongs to a personal `.gitignore` file 

:+1: 

There is nice article from GitHub: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

**Pull request merged**